diff --git a/config.def.h b/config.def.h
index a851bb6..671894a 100644
--- a/config.def.h
+++ b/config.def.h
@@ -10,6 +10,7 @@ static unsigned int gappov    = 30;       /* vert outer gap between windows and
 static       int smartgaps          = 0;        /* 1 means no outer gap when there is only one window */
 static int showbar            = 1;        /* 0 means no bar */
 static int topbar             = 1;        /* 0 means bottom bar */
+static Bool viewontag         = True;     /* Switch view on tag switch */
 static char font[]            = "monospace:size=10";
 static char dmenufont[]       = "monospace:size=10";
 static char *fonts[]          = { "monospace:size=10" };
diff --git a/dwm.c b/dwm.c
index a517be1..f5eb198 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1887,6 +1887,8 @@ tag(const Arg *arg)
 		selmon->sel->tags = arg->ui & TAGMASK;
 		focus(NULL);
 		arrange(selmon);
+		if(viewontag && ((arg->ui & TAGMASK) != TAGMASK))
+			view(arg);
 	}
 }
 
